cmd/api/src/api/v2/auth/sso_test.go

```diff
@@ -0,0 +1,265 @@
+// Copyright 2024 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package auth_test
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/gorilla/mux"
+	"github.com/specterops/bloodhound/src/api/v2/apitest"
+	"github.com/specterops/bloodhound/src/ctx"
+	"github.com/specterops/bloodhound/src/database/types/null"
+	"github.com/specterops/bloodhound/src/model"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+)
+
+func TestManagementResource_ListAuthProviders(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	resources, mockDB := apitest.NewAuthManagementResource(mockCtrl)
+	defer mockCtrl.Finish()
+
+	t.Run("successfully list auth providers without query parameters", func(t *testing.T) {
+		oidcProvider := model.OIDCProvider{
+			SSOProviderID: 1,
+			ClientID:      "client-id-1",
+			Issuer:        "https://issuer1.com",
+		}
+
+		samlProvider := model.SAMLProvider{
+			Serial:        model.Serial{ID: 2},
+			Name:          "SAML Provider 1",
+			DisplayName:   "SAML Provider One",
+			IssuerURI:     "https://saml-issuer1.com",
+			SSOProviderID: null.Int32From(2),
+		}
+
+		ssoProviders := []model.SSOProvider{
+			{
+				Serial:       model.Serial{ID: 1},
+				Name:         "OIDC Provider 1",
+				Slug:         "oidc-provider-1",
+				Type:         model.SessionAuthProviderOIDC,
+				OIDCProvider: &oidcProvider,
+			},
+			{
+				Serial:       model.Serial{ID: 2},
+				Name:         "SAML Provider 1",
+				Slug:         "saml-provider-1",
+				Type:         model.SessionAuthProviderSAML,
+				SAMLProvider: &samlProvider,
+			},
+		}
```

Comment:
might help cleanup but if nothing is being mutated inside of the t.Run blocks, you can safely move all the initialization steps above the t.Run() blocks